roxie/ccd/ccdserver.cpp

```diff
@@ -4557,8 +4579,30 @@ class CRemoteResultAdaptor : implements IEngineRowStream, implements IFinalRoxie
             //       But this could still cause too many reply packets on the fastlane
             //       (higher priority output Q), which may cause the activities on the 
             //       low priority output Q to not get service on time.
+
+            int origPriority = (int)ctx->queryOptions().priority;
+            int dynPriority = ctx->queryOptions().dynPriority;
+            if (dynPriority < origPriority)
+            {
+                unsigned newPri = getPriorityMask(dynPriority);
+                p->queryHeader().activityId &= ~ROXIE_PRIORITY_MASK;
+                p->queryHeader().activityId |= newPri;
+            }
+
+            // TODO: perhaps check elapsed every Nth msg ?
+            if ( (dynPriorityAdjustCycles > 0) && (origPriority == QUERY_LOW_PRIORITY_VALUE) && (dynPriority == QUERY_LOW_PRIORITY_VALUE) &&
+                 (ctx->queryElapsedCycles() > dynPriorityAdjustCycles) )
+            {
+                ctx->queryOptions().dynPriority = QUERY_BG_PRIORITY_VALUE;
+                unsigned dynAdjustMsec = (dynPriorityAdjustCycles * 1000ULL) / queryOneSecCycles();
+                UWARNLOG("WARNING: %d msec dynamic adjustment threshold reached, shifting query to BG queue", dynAdjustMsec);
+                p->queryHeader().activityId |= ROXIE_BG_PRIORITY;
```

Comment:
To be truly correct if the constants change value this should remove ROXIE_PRIORITY_MASK first.  Would it be better if this block was moved before the if() above?

Reply:
I agree, I can add a line so its -
+              p->queryHeader().activityId &= ~ROXIE_PRIORITY_MASK;
                p->queryHeader().activityId |= ROXIE_BG_PRIORITY;

Reply:
I think it would be better, but I am merging as-is for now.